thewalrus/csamples.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Classical Sampling Algorithms
+=============================
+
+.. currentmodule:: thewalrus.csamples
+
+This submodule provides access to classical sampling algorithms.
+
+Summary
+-------
+.. autosummary::
+    rescale_adjacency_matrix_thermal
+    rescale_adjacency_matrix
+    generate_thermal_samples
```

Comment:
One of the important design ideas behind `quantum.py` is that everything takes as argument a covariance matrix. Thus it does not make sense to move the functions in this module there. If anything it would make more sense to move hafnian_sample_graph and torontonian_sample_graph here.

Reply:
Great, how about we make a graph sampling module where the inputs are always graphs + any required utility functions?

Reply:
@josh146 suggested exactly the same thing a few minutes ago. Go for it!

Reply:
Although, all of this could be moved directly *into* the apps layer. As you can see it has *zero* dependency on the codebase of the walrus

Reply:
That's actually a good point. All the 'graph' functions do is simply convert to a covariance matrix and then call the covariance matrix sampler. They are essentially convenience wrappers. In addition, the functions in this module don't call any of the high performance C++ code, so they seem a bit tangential.

I guess a question in the back of my mind: is `thewalrus` the place for all sampling functions no matter what, or the place for high performance sampling algorithms that call underlying C++ libraries? For maintenance, the second would be preferred.

The gbs apps layer already contains its own sampling functions (monte carlo?) as well, I believe?